src/force/CabanaPD_Force_PMB.hpp

```diff
@@ -161,6 +161,43 @@ class Force<ExecutionSpace, ForceModel<PMB, Elastic, ModelParams...>>
             double w = 0.25 * model.c * s * s * xi * vol( j );
             W( i ) += w;
             Phi += w * vol( i );
+
+
+            
+            // Calculate forces for virial stress
+            double fx_i = 0.0;
```

Comment:
I had expected to use the current force, rather than recomputing - at least then you don't have to worry about them being zero. Unless I'm missing something it should just be the same way we currently interact with the volumes

Reply:
Ah I misunderstood you before, I thought we had to recalculate it. I've tried to slice the force outside of the kernel and in order to do that I had to create a const version of sliceForce which I have added to particles.hpp. 
In turn I have changes the internal logic of the virialstress calculations but it is still zero in the output of kalthoff winkler even though the force output has values.